File: JavaScript/5kyu/pete-the-baker.js
/**
 * Title: Pete, the baker
 * Link: https://www.codewars.com/kata/525c65e51bf619685c000059
 * Difficulty: 5 kyu
 *
 * ## Description
 *
 * Given a recipe object (ingredient -> amount per cake) and available amounts, return how many
 * whole cakes you can bake. Missing ingredients count as 0.
 */

function cakes(recipe, available) {
  let minCakes = Infinity;
  for (const [ingredient, need] of Object.entries(recipe)) {
    if (need === 0) continue;
    const have = Object.prototype.hasOwnProperty.call(available, ingredient)
      ? available[ingredient]
      : 0;
    const n = Math.floor(have / need);
    if (n < minCakes) minCakes = n;
  }
  return minCakes === Infinity ? 0 : minCakes;
}

File: Go/5kyu/pete-the-baker.go
// Title: Pete, the baker
// Link: https://www.codewars.com/kata/525c65e51bf619685c000059
// Difficulty: 5 kyu
//
// ## Description
//
// Given recipe and available ingredient amounts (maps), return how many whole cakes you can bake.
// Missing keys in available count as 0.

package kata

import "math"

func Cakes(recipe, available map[string]int) int {
	minCakes := math.MaxInt
	empty := true
	for ingredient, need := range recipe {
		if need == 0 {
			continue
		}
		empty = false
		have := available[ingredient]
		n := have / need
		if n < minCakes {
			minCakes = n
		}
	}
	if empty {
		return 0
	}
	return minCakes
}
